### ISSUE TITLE
Missing process pid and user data on ss-parser

### ISSUE THREAD

**Issue by gaugstein** [kellyjonbrazil/jc#388]
Hi, i miss the data for the process when i use the -e -o -p parameter. Example: `ss -4tulpn`

```bash
Netid              State                Recv-Q               Send-Q                               Local Address:Port                                Peer Address:Port                                                                                                                                                                                                   
udp                UNCONN               0                    0                                          0.0.0.0:4789                                     0.0.0.0:*                                                                                                                                                                                                      
udp                UNCONN               0                    0                                          0.0.0.0:54178                                    0.0.0.0:*                   users:(("snmpd",pid=1686,fd=12))                                                                                                                                                   
udp                UNCONN               0                    0                                          0.0.0.0:5353                                     0.0.0.0:*                   users:(("avahi-daemon",pid=1503,fd=12))                                                                                                                                            
udp                UNCONN               0                    0                                          0.0.0.0:7946                                     0.0.0.0:*                   users:(("dockerd",pid=3684,fd=27))                                                                                                                                                 
udp                UNCONN               0                    0                                          0.0.0.0:8125                                     0.0.0.0:*                  
```

Unfortunately the header data is missing when calling ss. But if you find a "user:((" in the last field it should be enough. Here is a small example of an implementation that can solve the problem:

```python
                if 'users:' in entry_list[-1]:
                    _field = entry_list[-1].rsplit(':', maxsplit=1)
                    _field = _field[1].replace('((','').replace('))', '').split('),(')
                    pids = {}
                    for item in _field:
                      _z = [ a.split('=') for a in item.split(',') ]
                      pids[_z[1][1]] = { "user": str(_z[0][0].strip('\"')), _z[2][0]: _z[2][1] }
                    header_list.append('process_pid')
                    entry_list[-1] = pids
```
If you add the part from line 383 in ss.py it should work and result in the following JSON:

```json
[
  {
    "netid": "udp",
    "state": "UNCONN",
    "recv_q": 0,
    "send_q": 0,
    "local_address": "0.0.0.0",
    "local_port": "4789",
    "peer_address": "0.0.0.0",
    "peer_port": "*",
    "local_port_num": 4789
  },
  {
    "netid": "udp",
    "state": "UNCONN",
    "recv_q": 0,
    "send_q": 0,
    "local_address": "0.0.0.0",
    "local_port": "54178",
    "peer_address": "0.0.0.0",
    "peer_port": "*",
    "process_pid": {
      "1686": {
        "user": "snmpd",
        "fd": "12"
      }
    },
    "local_port_num": 54178
  },
  {
    "netid": "udp",
    "state": "UNCONN",
    "recv_q": 0,
    "send_q": 0,
    "local_address": "0.0.0.0",
    "local_port": "5353",
    "peer_address": "0.0.0.0",
    "peer_port": "*",
    "process_pid": {
      "1503": {
        "user": "avahi-daemon",
        "fd": "12"
      }
    },
    "local_port_num": 5353
  },
  {
    "netid": "udp",
    "state": "UNCONN",
    "recv_q": 0,
    "send_q": 0,
    "local_address": "0.0.0.0",
    "local_port": "7946",
    "peer_address": "0.0.0.0",
    "peer_port": "*",
    "process_pid": {
      "3684": {
        "user": "dockerd",
        "fd": "27"
      }
    },
    "local_port_num": 7946
  },
....
``` 
Thanks so lot ..

**Comment by kellyjonbrazil** [comment]
Hi and thanks for reporting this issue. It is documented that the `-e` and `-o` options are not supported. I'll see if there is anything I can do here.

**Comment by gaugstein** [comment]
Hi, i have implemented a function for the missing options -e -o -p and have made some changes. Would you like to take a look at it? How should I send it to you, by commit change?
Kindly Regards

**Comment by kellyjonbrazil** [comment]
Nice - a Pull Request would be great. I'll take a look!

**Comment by gaugstein** [comment]
hope it worked ..

**Comment by gaugstein** [comment]
Curious, I have successfully tested the -a option on ubuntu 18. Do you have more information what could have gone wrong during the test?

**Comment by kellyjonbrazil** [comment]
The test is basically taking this `ss` output:
https://github.com/kellyjonbrazil/jc/blob/master/tests/fixtures/centos-7.7/ss-sudo-a.out

And comparing the `jc` output from that to this output that was generated by `jc` in the previous version:
https://github.com/kellyjonbrazil/jc/blob/master/tests/fixtures/centos-7.7/ss-sudo-a.json

```
First differing element 43:
{'net[69 chars]t': '* 0', 'path': '/run/systemd/notify 8971',[16 chars]': 0}
{'net[69 chars]t': '8971', 'peer_address': '*', 'peer_port': [74 chars]': 0}
```
This is showing that the 43'rd object seems to look different. The previous version of `jc` seemed to do the right thing, as you can see the local port looks right, while the new version of the parser parsing the local port as `* 0`, which seems to be incorrect.

**Comment by kellyjonbrazil** [comment]
Here is what element 43 looks like from the previous version of `jc`:
```
{
  "netid": "u_dgr",
  "state": "UNCONN",
  "recv_q": 0,
  "send_q": 0,
  "local_port": "8971",
  "peer_address": "*",
  "peer_port": "0",
  "path": "/run/systemd/notify",
  "local_port_num": 8971,
  "peer_port_num": 0
}
```

**Comment by gaugstein** [comment]
ok great, let me have a look tomorrow ..

**Comment by kellyjonbrazil** [comment]
It looks like the offending line in the `.out` file is 45:
```
% cat ss-sudo-a.out | grep -n 8971
45:u_dgr  UNCONN     0      0      /run/systemd/notify 8971                  * 0                    
148:u_dgr  UNCONN     0      0       * 14010                 * 8971       
```

**Comment by gaugstein** [comment]
tests integrated, please try

**Comment by kellyjonbrazil** [comment]
Looks good, thanks! I have merged this PR and it will be available in the next release. Thanks again for the contribution!

**Comment by kellyjonbrazil** [comment]
Released in v1.23.3
